components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts

```diff
@@ -406,10 +408,58 @@ export default function shieldsPanelReducer (
         console.error('Active tab not found')
         break
       }
+
+      const key = getAdblockCosmeticFiltersKey(action.tabId, action.frameId)
+      chrome.storage.local.get(key, (data = {}) => {
```

Comment:
Using `chrome.storage.local` for caching data introduces another overhead.
Need to find another way for caching data.